src/accelerate/state.py

```diff
@@ -437,11 +437,10 @@ def split_between_processes(self, inputs: list | tuple | dict | torch.Tensor, ap
             length = len(inputs[list(inputs.keys())[0]])
             if not all(len(v) == length for v in inputs.values()):
                 raise ValueError("All values in the dictionary must have the same length")
-        num_samples_per_process = math.ceil(length / self.num_processes)
-        start_index = self.process_index * num_samples_per_process
-        end_index = start_index + num_samples_per_process
-        if (len(inputs) % self.num_processes != 0) and (self.process_index == self.num_processes - 1):
-            end_index = length
+        num_samples_per_process = length // self.num_processes
+        num_extras = length % self.num_processes
```

Comment:
How about using the underappreciated `divmod` to calculate both in a single call:

```suggestion
        num_samples_per_process, num_extras = divmod(length, self.num_processes)
```